src/Database/FunctionsBuilder.php

```diff
@@ -127,10 +127,13 @@ public function coalesce(array $args, array $types = []): FunctionExpression
     }
 
     /**
-     * Returns a FunctionExpression representing a call to SQL CAST function.
+     * Returns a FunctionExpression representing a SQL CAST.
+     *
+     * The `$type` parameter is a SQL type. The return type for the returned expression
+     * is the default type name. Use `setReturnType()` to update it.
```

Comment:
Should we fix this caveat when using the new call signature? Seems like a good time to do it.

Reply:
I can't imply the return type from $type. I don't know how to fix the new signature.

Reply:
Are we not able to use a mapping of common SQL types and their PHP equivalents?

Reply:
All we have is the sql dialect column conversion. We don't have any sort of default mapping.

Reply:
Do we want to create a generic sql-type-to-database-type mapper?

Reply:
Probably not necessary given how frequently cast() is used it wouldn't be worth the investment.